lib/two_factor_authentication/models/two_factor_authenticatable.rb

```diff
@@ -109,6 +112,19 @@ def create_direct_otp(options = {})
 
         private
 
+        def reset_lockout
+          return unless self.class.lockout_reset_timeout.present?
+
+          if should_reset_lockout?
+            update_attributes(second_factor_attempts_count: 0, second_factor_attempts_locked_at: nil)
```

Comment:
`update_attributes` has been deprecated so think this would need to be `update`

https://apidock.com/rails/ActiveRecord/Base/update_attributes